packages/lib/gpc/src/gpcUtil.ts

```diff
@@ -215,6 +215,63 @@ export function makeWatermarkSignal(podValue: PODValue | undefined): bigint {
   return getPODValueForCircuit(podValue) ?? podValueHash(podValue);
 }
 
+/**
+ * General-purpose circuit requirements for a given proof.
+ * These values will be appropriately checked against the circuits at our
+ * disposal in order to accommodate all of the proof requirements. For the
+ * objects, entries, Merkle tree depth and membership list sizes, this amounts
+ * to picking the circuit description whose corresponding parameters exceed
+ * these numbers, while the choice of tuple parameters is more involved.
+ */
+export type GPCRequirements = {
+  /**
+   * Number of POD objects which can be included in a proof.
+   */
+  nObjects: number;
+
+  /**
+   * Number of POD entries which can be included in a proof.
+   */
+  nEntries: number;
+
+  /**
+   * Depth of POD merkle tree in the largest POD which can be included in a
+   * proof.  Max entries in any object is 2^(depth-1).
+   */
+  merkleMaxDepth: number;
+
+  /**
+   * Number of entries in each membership list to be included in proof.
+   * The order is the same as the list names (in alphabetical order).
+   */
+  nListElements: number[];
+
+  /**
+   * Arities (sizes) of tuples which can included in a proof.
+   * The order is the same as the tuple names (in alphabetical order).
```

Comment:
If you're going to rely on ordering by name, these should probably include the names directly, as `Record<PODName, number>`.  I think it's fine to leave it as-is until the next PR where the real needs will emerge, though.